### PR TITLE
[Artur Kostiuk]uit 69 loading on scroll

### PR DESCRIPTION
**Pull request by art1kos**
https://issues.griddynamics.net/browse/UIT-69

### CODE REVIEW

File: src/app/product-list-page/components/product-list-page/product-list-page.component.ts [hunk 1]
```diff
@@ -80,4 +90,15 @@ export class ProductListPageComponent implements OnInit, OnDestroy {
   private loadProducts(from: number, to: number): Observable<any> {
     return this.productsService.getProducts(`start=${from}&end=${to}`);
   }
+
+  public onScroll() {
+    const btnLoadMore = document.querySelector('.load-more');
```
Comment (dverbovyi) [inline src/app/product-list-page/components/product-list-page/product-list-page.component.ts]: you should track not a button, but the scroller position

File: src/app/product-list-page/components/product-list-page/product-list-page.component.ts [hunk 2]
```diff
@@ -43,6 +47,12 @@ export class ProductListPageComponent implements OnInit, OnDestroy {
     });
   }
 
+  ngAfterViewInit(): void {
+    if (this.loadMoreScrollEnabled) {
+      window.addEventListener('scroll', this.onScroll);
```
Comment (dverbovyi) [inline src/app/product-list-page/components/product-list-page/product-list-page.component.ts]: add debounce

**Review by dverbovyi — CHANGES_REQUESTED** [review]

**Review by dverbovyi — APPROVED** [review]
@art1kos please fix merge conflicts

**Comment by art1kos** [comment]
@dverbovyi  done

**Pull request merged**